### ISSUE TITLE
Updating hostnames automatically faster

### ISSUE THREAD

**Issue by samoht9277** [tailscale/tailscale#10780]
### What is the issue?

When setting up Tailscale on a server, the MagicDNS gets the device's hostname to be able to connect with the name of the machine instead of it's IP.

If I change the servers hostname, the server name stays old, updating after a really long time (weeks?).

It would be nice to see the hostname updating faster.

### Steps to reproduce

- Create a server.
- Install Tailscale.
- Check server's name on Tailscale's dashboard/CLI.
- Change the server's hostname.

### OS

Linux

### OS version

Debian 12

### Tailscale version

1.56.1

### Bug report

BUG-41229a8fa2405f97057dafd02cf37f456ca118d827767fd909af3cc3d467ada4-20240108161421Z-a83c2f94f1963bdf

**Comment by DentonGentry** [comment]
I imagine the new hostname is noticed the next time the Tailscale app starts up. There is no callback or notification when hostnames change.

**Comment by samoht9277** [comment]
Yeah, if you really need to see the hostname change, you can reauthorize the peer and it will update immediately.

`tailscale up --force-reauth`

**Comment by jayjb** [comment]
just an fyi: running `tailscale up` alone worked for me

**Comment by wjkoh** [comment]
> Yeah, if you really need to see the hostname change, you can reauthorize the peer and it will update immediately.
> 
> `tailscale up --force-reauth`

This works! Just make sure you remove the device from the admin console before running the command.
